File: app/src/main/java/com/example/progettoapp/YoutubeVideo.java
package com.example.progettoapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import com.google.android.youtube.player.YouTubeBaseActivity;
import com.google.android.youtube.player.YouTubeInitializationResult;
import com.google.android.youtube.player.YouTubePlayer;
import com.google.android.youtube.player.YouTubePlayerView;

public class YoutubeVideo extends YouTubeBaseActivity {

    Button playBtn;
    YouTubePlayerView youTubePlayerView;
    YouTubePlayer.OnInitializedListener onInitializedListener;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_youtube_video);

        Bundle intent = getIntent().getExtras();
        String videoID = intent.getString("videoID");
        String url = "https://www.youtube.com/watch?v="+videoID;

        Log.e("YoutubeVideo" ,"ID VIDEO="+videoID+"\n Url "+ url);

        playBtn = findViewById(R.id.playBtn);
        youTubePlayerView = findViewById(R.id.youtubeplayerview);
        onInitializedListener = new YouTubePlayer.OnInitializedListener() {
            @Override
            public void onInitializationSuccess(YouTubePlayer.Provider provider, YouTubePlayer youTubePlayer, boolean b) {
                youTubePlayer.loadVideo(videoID);
            }

            @Override
            public void onInitializationFailure(YouTubePlayer.Provider provider, YouTubeInitializationResult youTubeInitializationResult) {
            }
        };
        playBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                youTubePlayerView.initialize(getString(R.string.key),onInitializedListener);
            }
        });
    }
}

File: app/src/main/java/com/example/progettoapp/YoutubeVideo.kt
package com.example.progettoapp

import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import com.google.android.youtube.player.YouTubeBaseActivity
import com.google.android.youtube.player.YouTubeInitializationResult
import com.google.android.youtube.player.YouTubePlayer
import com.google.android.youtube.player.YouTubePlayerView

class YoutubeVideo : YouTubeBaseActivity() {
    var playBtn: Button? = null
    var youTubePlayerView: YouTubePlayerView? = null
    var onInitializedListener: YouTubePlayer.OnInitializedListener? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_youtube_video)
        val intent = intent.extras
        val videoID = intent!!.getString("videoID")
        val url = "https://www.youtube.com/watch?v=$videoID"
        Log.e("YoutubeVideo", "ID VIDEO=$videoID\n Url $url")
        playBtn = findViewById(R.id.playBtn)
        youTubePlayerView = findViewById(R.id.youtubeplayerview)
        onInitializedListener = object : YouTubePlayer.OnInitializedListener {
            override fun onInitializationSuccess(provider: YouTubePlayer.Provider, youTubePlayer: YouTubePlayer, b: Boolean) {
                youTubePlayer.loadVideo(videoID)
            }

            override fun onInitializationFailure(provider: YouTubePlayer.Provider, youTubeInitializationResult: YouTubeInitializationResult) {}
        }
        playBtn!!.setOnClickListener(View.OnClickListener { youTubePlayerView!!.initialize(getString(R.string.key), onInitializedListener) })
    }
}